src/main/groovy/org/cadixdev/gradle/licenser/LicenseExtension.groovy

```diff
@@ -58,21 +63,26 @@ class LicenseExtension extends LicenseProperties {
      * The charset to read/write the files with.
      * By default this is {@code UTF-8}.
      */
-    String charset = 'UTF-8'
+    final Property<String> headerCharset
```

Comment:
I think this may be better off as a breaking change, so that we can keep the original simpler names un-deprecated.

Reply:
Sounds reasonable -- I was hoping to avoid a compat break, but I suppose if there aren't sensible alternate names it's best to make a clean break.

Reply:
Having said that, will this impede groovy support at all?

Reply:
It shouldn't -- in fact i was surprised to find that the changes appear to be source-compatible across all the functional tests.

Gradle provides some syntactic sugar in Groovy DSL buildscripts that allows lazy properties to be set like normal variables -- though that doesn't exist in Kotlin buildscripts unfortunately.

Reply:
Does using the properties API make the Kotlin buildscript experience worse then, if I'm understanding you correctly?

Reply:
a little bit -- generally with properties what used to be (from Sponge):

```kotlin
license {
    (this as ExtensionAware).extra.apply {
        this["name"] = "SpongeAPI"
        this["organization"] = organization
        this["url"] = projectUrl
    }
    header = file("HEADER.txt")

    include("**/*.java")
    newLine = false
}
```

would turn into 

```kotlin
license {
    (this as ExtensionAware).extra.apply {
        this["name"] = "SpongeAPI"
        this["organization"] = organization
        this["url"] = projectUrl
    }
    header.set(resources.text.fromFile("HEADER.txt")) // or setHeader(file("HEADER.txt"))

    include("**/*.java")
    newLine.set(false)
}
```

What I've done in some projects is add extra methods to the extensions and tasks. I could do that for licenser, which would give:

```kotlin
license {
    (this as ExtensionAware).extra.apply {
        this["name"] = "SpongeAPI"
        this["organization"] = organization
        this["url"] = projectUrl
    }
    header(file("HEADER.txt"))

    include("**/*.java")
    newLine(false)
}
```

Reply:
Ok -- I've added those helper methods, and an additional functional test using the Kotlin buildscript for comparison.

Reply:
Okay - not as nice as before, but a consequence of Gradle's decisions 🤷